### PR TITLE
[Badge] Add critical styling

### PR DESCRIPTION
**Pull request by 416serg**
### WHY are these changes introduced?

Part of https://github.com/Shopify/polaris-react/issues/1969

Screen Shot 2020-04-07 at 10 10 48 AM

### WHAT is this pull request doing?

There was some discussion in a few issues:  https://github.com/Shopify/polaris-react/issues/1430 & https://github.com/Shopify/polaris-react/issues/1593

We have a few instances where we need to use the critical styling for the badge within Multi-Channel apps.

IE, currently, we have a product table within Google Shopping that uses Rails components and the `critical` badge exists in that environment

Screen Shot 2020-04-07 at 10 12 02 AM

We might be migrating it to React in the future, and I wanna have the ground covered. Additionally, we have a product sync status card that has a critical badge as well to get merchants to fix product errors

Screen Shot 2020-04-07 at 10 50 37 AM

## 

### How to 🎩

<details>
<summary>Copy-paste this code in <code>playground/Playground.tsx</code>:</summary>

```jsx
import React from 'react';
import {Badge} from '../src';

export function Playground() {
  return (
    <Badge status="critical">Error</Badge>
  );
}

```

</details>

### 🎩 checklist

* [x] Tested on [mobile](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting.md#cross-browser-testing)
* [x] Tested on [multiple browsers](https://help.shopify.com/en/manual/intro-to-shopify/shopify-admin/supported-browsers)
* [x] Tested for [accessibility](https://github.com/Shopify/polaris-react/blob/master/documentation/Accessibility%20testing.md)
* [x] Updated the component's `README.md` with documentation changes
* [x] [Tophatted documentation](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting%20documentation.md) changes in the style guide
* [ ] For visual design changes, pinged one of @ HYPD, @ mirualves, @ sarahill, or @ ry5n to update the Polaris UI kit

### CODE REVIEW

File: src/components/Select/Select.scss [hunk 1]
```diff
@@ -159,11 +163,6 @@ $stacking-order: (
     border-radius: var(--p-border-radius-base);
     background-color: var(--p-surface);
     @include focus-ring($border-width: rem(1px));
-
```
Comment (BPScott) [inline src/components/Select/Select.scss:162]: Revert this

**Comment by BPScott** [comment]
Ping @danrosenthal as I imagine he'll have opinions on color and role opinions

File: src/components/Badge/Badge.scss [hunk 2]
```diff
@@ -71,6 +71,17 @@ $pip-spacing: ($height - $pip-size) / 2;
   }
 }
 
+.statusCritical {
+  @include pip-color(var(--p-icon-warning, color('red', 'dark')));
```
Comment (BPScott) [inline src/components/Badge/Badge.scss:75]: This uses the same color as "warning" in the new design system - you'll need to update that

Comment (danrosenthal) [reply src/components/Badge/Badge.scss:75]:
```suggestion
  @include pip-color(var(--p-icon-critical, color('red', 'dark')));
```

File: src/components/Badge/Badge.scss [hunk 3]
```diff
@@ -71,6 +71,17 @@ $pip-spacing: ($height - $pip-size) / 2;
...
+  background-color: var(--p-surface-warning, color('red', 'light'));
```
Comment (danrosenthal) [inline src/components/Badge/Badge.scss:76]:
```suggestion
  background-color: var(--p-surface-critical, color('red', 'light'));
```

File: src/components/Badge/README.md [hunk 4]
```diff
@@ -182,12 +183,16 @@ Keep in mind that seeing this badge can feel stressful for merchants so it shoul
 
 ### Critical badge
 
-<!-- example-for: android, ios -->
+<!-- /content-for -->
```
Comment (danrosenthal) [inline src/components/Badge/README.md:186]: Why this change?

Comment (416serg) [reply src/components/Badge/README.md:186]: it wasn't showing up in story book without `/content-for` i think that's defined explicitly in the storybook setup

File: src/components/Select/Select.scss (hunk 1 above)
Comment (416serg) [reply src/components/Select/Select.scss:162]: yeah that was `yarn format` 😅sorry!

File: src/components/Badge/tests/Badge.test.tsx [hunk 5]
```diff
@@ -3,14 +3,28 @@ import React from 'react';
...
+      <Badge status="critical">Badge test</Badge>,
+    );
+    expect(
+      badge
+        .find('span')
+        .first()
+        .hasClass(styles[variationName('status', 'critical')]),
+    ).toBe(true);
+  });
+
```
Comment (416serg) [inline src/components/Badge/tests/Badge.test.tsx:27]: @danrosenthal I am having trouble coming up with an elegant test for `codecov` - let me know if this makes sense

File: src/components/Badge/README.md (hunk 4 above)
Comment (416serg) [reply src/components/Badge/README.md:186]: Looked through `polaris-readme-loader` is there a better way to surface this within storybook?

Comment (BPScott) [reply src/components/Badge/README.md:186]: Delete this line.

Note that the other examples that have contents for the web tab don't wrap it in anything.

File: src/components/Badge/tests/Badge.test.tsx (hunk 5 above)
Comment (BPScott) [reply src/components/Badge/tests/Badge.test.tsx:27]: Unit tests that check styles are applied aren't very valuable (there's no guarantee that style exists or does any coloring).

Changes in this will be caught by the visual regression tests that run over our examples - and you've now added one of them in storybook.

I'd be tempted to leave this out.

File: src/components/Badge/README.md (hunk 4 above)
Comment (416serg) [reply src/components/Badge/README.md]: makes sense, removing the comment makes it available for all examples, thank you!

File: src/components/Badge/tests/Badge.test.tsx (hunk 5 above)
Comment (416serg) [reply src/components/Badge/tests/Badge.test.tsx:27]: leaving it out will fail `codecov` - is that okay in this instance?

Comment (BPScott) [reply src/components/Badge/tests/Badge.test.tsx:27]: Yeah I'm fine with that :)

Comment (416serg) [reply src/components/Badge/tests/Badge.test.tsx]: done

**Comment by 416serg** [comment]
@danrosenthal translations are all in 😄

**Comment by BPScott** [comment]
Looks like that translations stuff was down to a slightly goofy rebase / merge.

@416serg you can't merge this yourself thanks to the codecov check but I can flex my repo admin muscles as I've said that's not a big deal. Can you confirm this is good to merge?

**Comment by 416serg** [comment]
Yeah, the translations broke from a goofy rebase but aren’t part of this PR
- were good to go here 🚢

**Comment by BPScott** [comment]
Thanks for confirming!
